samples/bluetooth/alexa_gadget/src/gadgets/CMakeLists.txt

```diff
@@ -0,0 +1,192 @@
+#
+# Copyright (c) 2020 Nordic Semiconductor
+#
+# SPDX-License-Identifier: LicenseRef-BSD-5-Clause-Nordic
+#
+
+# The Gadgets Profile relies on the following external dependencies that
+# must be downloaded and run before the build stage:
+# - nanopb generator
+# - protocol buffer definitions
+
+# Note: as there doesn't seem to be an easy way of checking the version of the nanopb generator (protoc-gen-nanopb binary)
+# there is the possibility of a version mismatch between the generator and the nanopb source files that is used in the firmware sample.
+# If there is a mismatch, there will be a compilation error in the build step.
+# The easiest way to fix such an error is to download the appropriate nanopb generator .zip and extract it to this sample folder.
+# (https://jpa.kapsi.fi/nanopb/download/)
+
+# Check local folder for protoc and nanopb binaries presence
+file(GLOB_RECURSE LOCAL_PROTOC_BIN "${PROJECT_SOURCE_DIR}/*/generator-bin/protoc")
+if (NOT LOCAL_PROTOC_BIN)
+    file(GLOB_RECURSE LOCAL_PROTOC_BIN "${PROJECT_SOURCE_DIR}/*/generator-bin/protoc.exe")
+endif()
+file(GLOB_RECURSE LOCAL_NANOPB_BIN "${PROJECT_SOURCE_DIR}/*/generator-bin/protoc-gen-nanopb*")
+
+# Check system for protoc and nanopb binaries presence
+find_program(PROTOC_BIN protoc)
+find_program(NANOPB_BIN protoc-gen-nanopb)
+
+# Prefer local binaries to make version matching with nanopb source repo easier
+if (LOCAL_PROTOC_BIN)
+    set(PROTOC_BIN ${LOCAL_PROTOC_BIN})
+endif()
+
+if (LOCAL_NANOPB_BIN)
+    set(NANOPB_BIN ${LOCAL_NANOPB_BIN})
+endif()
+
+if ((NOT NANOPB_BIN) OR (NOT PROTOC_BIN))
+    message(STATUS "nanopb not found.")
+    message(STATUS "Please download nanopb from here https://jpa.kapsi.fi/nanopb/download/")
+    message(STATUS "and extract to either this sample directory or somewhere cmake can find it via system PATH.")
```

Comment:
The problem is that the protobuf compilation is done by the protoc-compiler binary, which invokes nanopb as a plugin. The nanopb github repo doesn't include platform-specific protoc binaries, but the downloads from https://jpa.kapsi.fi/nanopb/download/ does. E.g. the windows .zip download includes protoc.exe

Reply:
I see now, that there is a difference between:
https://jpa.kapsi.fi/nanopb/download/nanopb-0.4.2-linux-x86.tar.gz
and 
https://jpa.kapsi.fi/nanopb/download/nanopb-0.4.2.tar.gz

and I of course downloaded `nanopb-0.4.2.tar.gz` (which is identical to the repo).

Consider if this can be made clearer, cause I bet there will be end-users ding the same.

Reply:
Yeah, good point. Will clarify in the documentation